src/config/mod.rs

```diff
@@ -896,6 +896,9 @@ impl<'a> de::Deserialize<'a> for ActionWrapper {
                     "Quit" => Action::Quit,
                     "ClearLogNotice" => Action::ClearLogNotice,
                     "SpawnNewInstance" => Action::SpawnNewInstance,
+                    "ToggleFullscreen" => Action::ToggleFullscreen,
+                    #[cfg(target_os = "macos")]
+                    "ToggleSimpleFullscreen" => Action::ToggleSimpleFullscreen,
```

Comment:
In terms of Apple's guidelines, they recommend only the "native" fullscreen which occupies its own space, and is quite obtrusive.

I implemented both here and made the both available because in #34 @jwilm suggests that [he wants the "simple" fullscreen mode](https://github.com/jwilm/alacritty/issues/34#issuecomment-271131103):

> At least on macOS, there's the automagical full screen button that does the trick. It does occupy another workspace, however, and it doesn't allow stacking the full screen application on top of other apps.
>
> That's precisely what this issue is for :)

And I've left in the "native" fullscreen mode so that Alacritty can follow Apple's guidelines and also have the standard fullscreen mode.

Reply:
Is there any plan to deprecate the old fullscreen mode? If Apple doesn't want to support it anymore, it seems like that needs to addressed with them and not in Alacritty.

Also is there any requirement to add keybindings for both, or is there already a default keybinding in macos that toggles the new fullscreen mode anyways?

Reply:
Sorry, I was wrong to assume you were familiar with the fullscreen situation on macOS, let me explain:

Basically, macOS has three distinct fullscreen modes:

* Standard fullscreen (happens when you click the green fullscreen button)
    * Also called "native" fullscreen
    * This moves the window to its own space
    * There's no control over the animation or transition - all handled by macOS itself
    * This is the most "recent" of the fullscreen modes and was added to OSX around version 10.7
* [`enterFullScreenMode`](https://developer.apple.com/documentation/appkit/nsview/1483780-enterfullscreenmode?language=objc)
    * This takes over the monitor, so _no spaces or other applications may be used at the same time_ (think of how fullscreen mode works in kiosk applications)
    * Also completely controlled by macOS itself (ie: unconfigurable)
* "Simple Fullscreen"
    * The window just fills the screen completely (hiding the dock and menu bar)
    * This used to be how all fullscreen windows worked pre-macOS Lion
    * This is a popular feature, so much so that many apps have implemented it manually since Apple removed the functionality.
        * Some apps are: VLC, iTerm2, [The `electron` framework](https://electronjs.org/docs/api/browser-window#winsetsimplefullscreenflag-macos)
        * It's also worth noting that web browsers (Chrome, Firefox, Opera, etc) also implement this type of fullscreen mode when viewing videos and other such media
    * This means you can still use other applications, spaces, and all other functionality, while having your application in fullscreen mode.
    * This fullscreen mode is most similar to the fullscreen modes of Windows and Linux (using Gnome, etc).

So, there's no "deprecated" fullscreen mode, and macOS defaults to setting <kbd>ctrl</kbd> + <kbd>cmd</kbd> + <kbd>F</kbd> as the Standard Fullscreen option.